lib/agent/agent.vala

```diff
@@ -1709,6 +1715,23 @@ namespace Frida.Agent {
 			base.dispose ();
 		}
 
+		private static int replacement_task_threads (uint task_id, uint ** threads, uint * count) {
+			unowned Gum.InvocationContext context = Gum.Interceptor.get_current_invocation ();
+			unowned ThreadSuspendMonitor monitor = (ThreadSuspendMonitor) context.get_replacement_function_data ();
+
+			return monitor.handle_task_threads (task_id, threads, count);
+		}
+
+		private int handle_task_threads (uint task_id, uint ** threads, uint * count) {
+			int result = task_threads (task_id, threads, count);
+
+			_remove_cloaked_threads (task_id, threads, count);
+
+			return result;
+		}
+
+		public extern void _remove_cloaked_threads (uint task_id, uint ** threads, uint * count);
```

Comment:
I guess this can be a static method?

```suggestion
		public extern static void _remove_cloaked_threads (uint task_id, uint ** threads, uint * count);
```

Reply:
right! ❤️